bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -258,6 +258,10 @@ public static boolean anyToJBoolean(Object sourceVal) {
      * @return true if the value belongs to the given type, false otherwise
      */
     public static boolean checkIsType(Object sourceVal, Type targetType) {
```

Comment:
This will impact on all checkIsType calls. Given issue only occurs for interop methods returning union types. Can we handle onlly those.